src/main/java/frc/robot/arm/ArmSubsystem.java

```diff
@@ -0,0 +1,110 @@
+package frc.robot.arm;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Robot;
+import frc.robot.Robot.RobotType;
+import java.util.function.Supplier;
+import org.littletonrobotics.junction.Logger;
+
+public class ArmSubsystem extends SubsystemBase {
+  // constants
+  // TODO: change to real values
+  public static final double PIVOT_RATIO = 10;
+  public static final Rotation2d MAX_ANGLE = Rotation2d.fromDegrees(180);
+  public static final Rotation2d MIN_ANGLE = Rotation2d.fromDegrees(0);
+
+  private ArmIO io;
+  private ArmIOInputsAutoLogged inputs = new ArmIOInputsAutoLogged();
+
+  private ArmState armState = ArmState.IDLE;
+  private IntakeState intakeState = IntakeState.IDLE;
+
+  public ArmSubsystem(ArmIO io) {
+    this.io = io;
+  }
+
+  @Override
+  public void periodic() {
+    io.updateInputs(inputs);
+    Logger.processInputs("arm", inputs);
+  }
+
+  public Command setTargetAngle(Rotation2d target) {
+    return setTargetAngle(() -> target);
+  }
+
+  public Command setTargetAngle(Supplier<Rotation2d> target) {
+    return this.runOnce(
+            () -> {
+              if (Robot.ROBOT_TYPE != RobotType.REAL) Logger.recordOutput("arm", target.get());
+            })
+        .andThen(this.run(() -> io.setMotorPosition(target.get())));
```

Comment:
```suggestion
    return this.runOnce(
            () -> {
              if (Robot.ROBOT_TYPE != RobotType.REAL) Logger.recordOutput("Arm", target.get());
              io.setMotorPosition(target.get())
            });
```
The `this.run()` would run the `setMotorPosition` method forever, which would then block everything else. so we can include it in the `runOnce()` with the logging statement. also make sure you have consistent capitalization